sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -4721,13 +4721,91 @@ def repeat(self, repeats):
       pd.core.strings.StringMethods, 'get_dummies',
       reason='non-deferred-columns')
 
-  split = frame_base.wont_implement_method(
-      pd.core.strings.StringMethods, 'split',
-      reason='non-deferred-columns')
+  def _split_helper(
+    self, rsplit=False, pat=None, expand=False, regex=None, **kwargs):
+
+    # Adding arguments to kwargs. regex introduced in pandas 1.4
+    # but only for split, not rsplit
+    kwargs['pat'] = pat
+    kwargs['expand'] = expand
+    if PD_VERSION >= (1, 4) and not rsplit:
+      kwargs['regex'] = regex
```

Comment:
Odd that `regex` is in `split` only, that's frustrating.

It _might_ be cleaner to just pass through all the arguments via `**kwargs`, and pull out the ones you need to branch on.

Reply:
I need to utilize all the args for branching logic, so it seems to be equally clean/unclean. I can simply the function definition, and pretty much do the reverse with the logic. Do you think this is syntactically better?
```
    pat = kwargs.get('pat', None)
    expand = kwargs.get('expand', False)
    regex = kwargs.get('regex', None)
    if PD_VERSION < (1, 4) and not rsplit:
      kwargs.pop('regex', None)
```

Reply:
I pushed a commit to incorporate all your suggestions, including one in the above comment. But we can change that one back if stylistically, the original is better.

Reply:
I think we should be able to drop the PD_VERSION check at least. The arg doesn't exist prior to 1.4 so it can't be in kwargs.